### DISCUSSION TITLE
How to add type for a argument that has an array as a default value?

### DISCUSSION THREAD

**Discussion by exts** [rectorphp/rector#8424] [Q&A] [answered]
# Question
How do I get rector to generate array argument parameters for array's that are default values example:

```php
public function example($array = []) 
{
    return $array;
}
```

```php
public function example(array $array = []) : array
{
    return $array;
}
```

**Comment by samsonasik** [comment, marked as answer]
You can use `ArgumentAdderRector`, and enable `TYPE_DECLARATION` set:, see 

https://getrector.com/demo/33ae3161-c578-4b54-b601-fe32f304ba48

**Comment by exts** [comment]
oh wow lol okay I think I have a bit of source reading to understand what's happening

Thanks

**Reply by cyppe** [reply]
I also needed this..  What is the reason this is not a default rule? Seems like adding array in such obvious case should be common, but I probably are missing something.
